bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/MyWarmupAccountHandler.java

```diff
@@ -0,0 +1,142 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class MyWarmupAccountHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MyWarmupAccountHandler.class);
+    private final MyWarmupApi api;
+
+    private @Nullable QueryResponseDTO queryResponse = null;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable WarmupDiscoveryService discoveryService;
+
+    public MyWarmupAccountHandler(Bridge thing, final HttpClient httpClient) {
+        super(thing);
+        api = new MyWarmupApi(httpClient, getConfigAs(MyWarmupConfigurationDTO.class));
+    }
+
+    @Override
+    public void initialize() {
+        MyWarmupConfigurationDTO config = getConfigAs(MyWarmupConfigurationDTO.class);
+        if (config.username.length() == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Username not configured");
+        } else if (config.password.length() == 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Password not configured");
+        } else if (config.refreshInterval >= 10) {
+            api.setConfiguration(config);
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshFromServer, 0, config.refreshInterval,
+                    TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Refresh interval misconfigured (minimum 10s)");
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(WarmupDiscoveryService.class);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        cancelRefresh();
+    }
+
+    public void cancelRefresh() {
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    public synchronized void refreshFromServer() {
+        try {
+            queryResponse = api.getStatus();
+        } catch (MyWarmupApiException e) {
+            queryResponse = null;
+            logger.debug("{}", e.getMessage(), e);
+        } finally {
+            if (queryResponse != null) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to contact MyWarmup");
+            }
```

Comment:
It seems a bit unreasonable to log the stack trace when the network fails. You could also include the exception's message in the ThingStatusDetailMessage.

Reply:
Hi @fwolter , I've tweaked the error handling again, and made the changes to sync up with v3 changes - hope this is better!